### PR TITLE
Add UI Tests

### LINKED ISSUE #6: End to end testing
There should be some end to end testing, ideally with galata: https://github.com/jupyterlab/galata

### PR DESCRIPTION
**Pull request by jtpio**
## References

Fixes #6 
Replaces https://github.com/jupyterlite/jupyterlite/pull/258

A reboot of #258 since the Galata setup and API have changed a lot recently, and Galata is now part of the JupyterLab main repo.

## Code changes

Most of the changes are self-contained in the `ui-tests` folder. The structure and setup should be similar to the other repos using galata.

### TODO

- [x] Basic UI tests
- [x] Test on `firefox` and `chromium`
- [x] Setup visual regression with reference snapshots
- [x] Test with initial server contents (folders, subfolders, with spaces)
- [x] Move the build step to a reusable action
- [x] Document how to update snapshots (in the contributing guide)

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

File: tsconfig.eslint.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "extends": "./tsconfigbase",
-  "include": ["packages/**/*", "app/**/*"],
+  "include": ["packages/**/*", "app/**/*", "ui-tests"],
```
Comment (bollwyvl) [inline tsconfig.eslint.json:3]: is this going to make things even worse re OoM?

Comment (jtpio) [reply tsconfig.eslint.json:3]: Probably yes, unfortunately I haven't been able to run ESLint on this repo for a while now... (still hitting #194 almost consistently)

**Comment by jtpio** [comment]
This can be convenient to quickly try something out on the demo site:

https://user-images.githubusercontent.com/591645/140836061-00fe8e25-f2d1-40dd-a317-1d783fcbf0d8.mp4

**Comment by jtpio** [comment]
I think this should be good enough of a start for now to get the basic structure for ui tests.

There are a few tests that should now cover the contents manager a bit more, since we had a couple of regressions related to that recently.

The structure is very similar to the one in upstream JupyterLab: https://github.com/jupyterlab/jupyterlab/tree/master/galata
So this should make it easier for folks familiar with that setup.

Possible follow-ups:

- test widgets and outputs render correctly in the kernels
  - maybe we get #310 in first
  - let's see if we want to do that in this repo or not (#386)
- more test to catch missing plugins / features when updating to newer packages

**Pull request merged**